crates/config/src/lib.rs

```diff
@@ -166,7 +166,7 @@ pub async fn load_config() -> anyhow::Result<LuaConfig> {
         };
 
         let _timer = latency_timer("context-creation");
-        func.call_async(()).await?;
+        func.call_async::<_, ()>(()).await?;
```

Comment:
```rust
warning: this function depends on never type fallback being `()`
   --> crates/config/src/lib.rs:119:1
    |
119 | pub async fn load_config() -> anyhow::Result<LuaConfig> {
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
    = note: for more information, see issue #123748 <https://github.com/rust-lang/rust/issues/123748>
    = help: specify the types explicitly
note: in edition 2024, the requirement `!: FromLua<'_>` will fail
   --> crates/config/src/lib.rs:169:14
    |
169 |         func.call_async(()).await?;
    |              ^^^^^^^^^^
    = note: `#[warn(dependency_on_unit_never_type_fallback)]` on by default
```